### ISSUE TITLE
 Does it work with Angular?

### ISSUE THREAD

**Issue by blasco** [craigbuckler/progressive-image.js#36]
I can't make it work, is it not supported?

**Comment by craigbuckler** [comment]
It's vanilla JavaScript so it *should* work with any framework.

However, you need to ensure Angular uses the correct HTML and classes when writing content to the page.

Is the generated HTML correct? Are you seeing errors in the console?

**Comment by blasco** [comment]
I can't manage to make it work even with a simple html file:

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/progressive-image.js/dist/progressive-image.css">
<script src="https://cdn.jsdelivr.net/npm/progressive-image.js/dist/progressive-image.js"></script>
  </head>
  <body>
        <a href="assets/img/AMSTERDAM.jpg" class="boxout2 progressive replace">
          <img src="assets/loading.jpg" class="preview" loading="lazy" width="32" height="24" alt="cave" />
        </a>
  </body>
</html>
```

I can see the blurred image, and when I click it it opens the high resolution image, but only when I click it (same behavior with angualr) not when the page loads or the image enters the viewport. Am I using the library wrong?

**Comment by craigbuckler** [comment]
Place the `<script>` just before the closing `<body>` tag and it'll work.

That said, it should have worked anyway. I'll fix the problem.

**Comment by blasco** [comment]
You are right, it works after applying your proposed change. Great! What is the issue and how will you solve it? (out of curiosity)

**Comment by craigbuckler** [comment]
I think it's because it's checking `document.body` is available before it is!

That only occurs when `<script>` tags are loaded in the `<head>`. Developers rarely do that, so it's not been spotted until now!
